### PR TITLE
ci(docker): Overhaul Docker stuff

### PR DESCRIPTION
**Pull request by BYK**
- Switch to 2-step build process for Docker images, reducing build times and final runtime image size
- Add `:latest` image publishing to Craft

### CODE REVIEW

**Comment by BYK** [comment]
@EvanPurkhiser getting `Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"`, not sure if this is normal or not.

**Comment by EvanPurkhiser** [comment]
Yeah I've seen that as well, not sure if it's going to be a problem or not yet.

File: Dockerfile [hunk 1]
```diff
@@ -15,8 +17,5 @@ COPY package.json yarn.lock ./
 RUN yarn install
```
Comment (EvanPurkhiser) [inline Dockerfile:17]: In that case I think we only need to install the runtime dependencies, so we can probably add in `--production`

Comment (BYK) [reply Dockerfile:17]: The added `ENV NODE_ENV=production` above covers this case.

Comment (EvanPurkhiser) [reply Dockerfile:17]: cool 👍

**Review by EvanPurkhiser — APPROVED** [review]

**Pull request merged**